test/NuGet.Core.Tests/NuGet.Resolver.Test/ResolverUtilityTests.cs

```diff
@@ -314,6 +314,62 @@ public void ResolverUtility_GetDiagnosticMessageForIncompatibleTargetWithAllowed
             Assert.Equal("Unable to find a version of 'b' that is compatible with 'a 1.0.0 constraint: b (= 1.0.0)'. 'b' has an additional constraint (= 2.0.0) defined in packages.config.", message);
         }
 
+        public void ResolverUtility_GetDiagnosticMessageForIncompatibleDependencyWithAllowedVersion()
```

Comment:
Add tests for removal of *, and for non normalized input strings (like 1.1, and 1.1.0.0) both in package version and dependencies